### PR TITLE
Add filesystem argument for reading DeltaTable in Python binding

### PR DESCRIPTION
**Pull request by fvaleye**
# Description
- Expose `filesystem` argument for reading DeltaTable with Pyarrow in the Python binding

# Related Issue(s)
#392

### CODE REVIEW

**Review by houqp — APPROVED** [review]
looks like there is a code conflict

File: python/deltalake/table.py [hunk 1]
```diff
@@ -190,15 +191,16 @@ def pyarrow_schema(self) -> pyarrow.Schema:
         return pyarrow_schema_from_json(self._table.arrow_schema_json())
 
     def to_pyarrow_dataset(
-        self, partitions: Optional[List[Tuple[str, str, Any]]] = None
+        self, partitions: Optional[List[Tuple[str, str, Any]]] = None, filesystem: FileSystem = None
```
Comment (houqp) [inline python/deltalake/table.py:194]: The type annotation here should be `Optional[FileSsytem]`

**Review by houqp — APPROVED** [review]

**Comment by mattc-eostar** [comment]
I am not sure that this addresses the specific issue with how ADLS needs a container prepended to the file names for use within the file system object as described the issue related to this (#392).

**Comment by prasadvaze** [comment]
@mattc-eostar  is correct. We still need to prepend container-name to the file names. So the issue is still open.  @fvaleye 

In the meantime tried using blobfuse instead of adlfs to connect and mount ADLS on ubuntu. The performance is much better (2 sec versus 6 sec to query ~50k rows delta table.)  And not required to use to_pyarrow_dataset( )

**Pull request merged**